lib/fs/basicfs_watch.go

```diff
@@ -74,7 +79,18 @@ func (f *BasicFilesystem) watchLoop(name, evalRoot string, backendChan chan noti
 
 		select {
 		case ev := <-backendChan:
-			relPath := f.unrootedChecked(ev.Path(), evalRoot)
+			relPath, err := f.unrootedChecked(ev.Path(), evalRoot)
+			if err != nil {
+				select {
+				case errChan <- err:
+					l.Debugln(f.Type(), f.URI(), "Watch: Sending error", err)
+				case <-ctx.Done():
+				}
+				notify.Stop(backendChan)
+				l.Debugln(f.Type(), f.URI(), "Watch: Stopped due to", err)
+				return
```

Comment:
Who closes outChan?
I don't see anyone closing it anywhere.

Reply:
That's correct and actually intended, as the context is the stopping/unblocking mechanism. This allows the receiver to use whatever it gets from the channel without checking for `nil`(or getting/checking the second bool "closedness" value). Is there some pitfall I am not seeing with not closing channels? I only ever know about the (somewhat obvious) principle of only sender should close a channel, but not that channels should be closed.

Reply:
Isn't the aggregator just
```
go func() {
   for x := range chan {
   }
}()
```
?
That doesn't need to check for closedness and does not leak a routine (it probably doesn't anyway as channels are closed when gced, and where there are no writers left)

Reply:
Well but something happens inside the loop, and if `x` is not a real event but either `nil` or a default struct behaviour is not defined. And as in almost everywhere it's:

```
go func() {
   for {
      var x
      select {
         case x = <-chan:
         case <-ctx.Done():
            return
      }
   }
   ...
}()
```

Reply:
Fine